src/hotspot/share/classfile/klassFactory.cpp

```diff
@@ -216,7 +216,7 @@ InstanceKlass* KlassFactory::create_from_stream(ClassFileStream* stream,
 
 #if INCLUDE_CDS
   if (Arguments::is_dumping_archive()) {
-    ClassLoader::record_result(result, stream, THREAD);
+    ClassLoader::record_result(THREAD, result, stream);
```

Comment:
There are many non-exception related uses of THREAD in this method. This seems like a candidate for introducing:
JavaThread* current = THREAD->as_Java_thread()
and using "current" throughout instead of THREAD in those non-exception cases.

Reply:
I've made the changes as you suggested.

Reply:
Just for completeness I reverted my position on this due to objections from Coleen about this kind of change (made elsewhere). So we stick with THREAD here.